### ISSUE TITLE
Bug: [no-empty-object-type] suggestion can produce invalid code

### ISSUE THREAD

**Issue by baevm** [typescript-eslint/typescript-eslint#12736]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.
- [x] If I used AI assistance to prepare this issue, I have reviewed and followed the [AI Contribution Policy](https://typescript-eslint.io/contributing/ai-policy/).

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&showAST=es&fileType=.tsx&code=KYDwDg9gTgLgBAE2AMwIYFcA28CWA7GYKNAY2DgDEII4BvOAXwCgX9DjUy4AhVKOxqwJFS5Xv3oA3VJnTAAXHADOMKPgDmjIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge3oFsbDaOAIwBWiMnSKl0YANqRE0aB2iQAuuDABfEJqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
// Example 1
export default interface Foo { }


// Example 2
interface Bar { }

interface Bar { value: string }
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-empty-object-type": ["error"]
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

In Example 1, it should not suggest a fix for an empty interface when the interface is wrapped in an ExportDefaultDeclaration

In Example 2, it should not suggest a fix if the interface participates in declaration merging with another interface

### Actual Result

In Example 1, the code is fixed to:
```
export default type Foo = object
```

This produces invalid TypeScript syntax (https://github.com/microsoft/TypeScript/issues/41409).

In Example 2, the code is fixed to:
```
type Bar = object

interface Bar { value: string }
```

This produces the following TypeScript error: `Duplicate identifier 'Bar'.`

**Comment by StyleShit** [comment]
Definitely buggy, thanks for reporting!
